gson/src/main/java/com/google/gson/internal/LinkedTreeMap.java

```diff
@@ -43,13 +43,7 @@
 @SuppressWarnings("serial") // ignore warning about missing serialVersionUID
 public final class LinkedTreeMap<K, V> extends AbstractMap<K, V> implements Serializable {
   @SuppressWarnings({"unchecked", "rawtypes"}) // to avoid Comparable<Comparable<Comparable<...>>>
-  private static final Comparator<Comparable> NATURAL_ORDER =
-      new Comparator<Comparable>() {
-        @Override
-        public int compare(Comparable a, Comparable b) {
-          return a.compareTo(b);
-        }
-      };
+  private static final Comparator<Comparable> NATURAL_ORDER = Comparator.naturalOrder();
```

Comment:
The code below performs identity checks (`== NATURAL_ORDER`) on this, but I hope this should be fine.